### PR TITLE
Feature/version-one API

### PR DESCRIPTION
**Pull request by defra-dom**
- add v1 data-source and mocks
- add v1 datasource/transformers/mock endpoints and update resolvers

### CODE REVIEW

File: app/auth/authenticate.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import logger from '../utils/logger.js'
 
 export async function getJwtPublicKey (kid) {
   const client = jwksClient({
-    jwksUri: `https://login.microsoftonline.com/${process.env.API_TENANT_ID}/discovery/v2.0/keys`
+    jwksUri: `https://login.microsoftonline.com/${process.env.RP_INTERNAL_APIM_TENANT_ID}/discovery/v2.0/keys`
```
Comment (jack-mccabe-defra) [inline app/auth/authenticate.js:8]: This is not right, this is authentication for our API and should not change

Comment (defra-dom) [reply app/auth/authenticate.js]: My mistake, global search and replace. think ive sorted it now....

File: app/data-sources/version-one/version-one.js
Comment (jack-mccabe-defra) [inline app/data-sources/version-one/version-one.js:1]: Can we avoid using version one in any naming, this gateway we are accessing is called rp internal so maybe could name it like that. Just cautious Version One could lead to some confusion espexially if they roll off as a supplier.

File: test/resolvers/permissions.test.js [hunk 2]
```diff
@@ -67,11 +65,10 @@ test('Permission.active', async () => {
...
     },
-    { customerId: 'mockCustomerId', businessId: 'mockBusinessId' },
+    { crn: '1102634220', sbi: '107183280' },
     {
       dataSources
     }
   )
-  // TODO this should return true
-  expect(response).toEqual(false)
+  expect(response).toEqual(true)
```
Comment (rkdefra) [inline test/resolvers/permissions.test.js:73]: :tada:

**Review by rkdefra — CHANGES_REQUESTED** [review]
A couple of minor things to check, but otherwise looks ok.
Can we please have multiple smaller PRs instead of mega-changes in the future :pray:

File: appConfig/appConfig.snd4.yaml [hunk 3]
```diff
@@ -1,34 +1,50 @@
...
-- key: RURAL_PAYMENTS_PORTAL_EMAIL
-  value: 'example@email.com'
+  value: "2"
 - key: RURAL_PAYMENTS_PORTAL_API_URL
-  value: 'http://127.0.0.1:3100/rpp/'
-- key: RURAL_PAYMENTS_PROXY_URL
-  value: ''
+  value: "http://127.0.0.1:3100/rpp/"
+- key: LOG_LEVEL
+  value: "debug"
```
[the remark this answers is not in the document]
Comment (jack-mccabe-defra) [reply appConfig/appConfig.snd4.yaml:6]: it is dev data but you are right I was just checking some things so will revert this

File: app/transformers/rural-payments/customer.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  return notifications
+    .map(message => ({
+      id: message.id,
+      title: message.title,
+      date: message.createdAt,
+      body: message.body,
+      read: !!message.readAt,
+      archivedAt: message.archivedAt
+    }))
+    .filter(({ archivedAt }) =>
```
[the remark this answers is not in the document]
Comment (jack-mccabe-defra) [reply app/transformers/rural-payments/customer.js:121]: it would

**Review by rkdefra — APPROVED** [review]

**Review by rkdefra — APPROVED** [review]
Looks good!

**Pull request merged**